sdk/identity/azure_identity/src/lib.rs

```diff
@@ -103,13 +104,33 @@ where
 const AZURE_AUTHORITY_HOST_ENV_KEY: &str = "AZURE_AUTHORITY_HOST";
 const AZURE_PUBLIC_CLOUD: &str = "https://login.microsoftonline.com";
 
-fn get_authority_host(env: Option<Env>, option: Option<String>) -> Result<Url> {
-    let authority_host = env
-        .unwrap_or_default()
-        .var(AZURE_AUTHORITY_HOST_ENV_KEY)
-        .unwrap_or_else(|_| option.unwrap_or_else(|| AZURE_PUBLIC_CLOUD.to_owned()));
+fn get_authority_host(env: Option<Env>, cloud: Option<&CloudConfiguration>) -> Result<Url> {
+    let authority_host = match cloud {
+        None => env
+            .unwrap_or_default()
+            .var(AZURE_AUTHORITY_HOST_ENV_KEY)
+            .unwrap_or_else(|_| AZURE_PUBLIC_CLOUD.to_string()),
+        Some(CloudConfiguration::Custom(config)) => config.authority_host.clone(),
+        Some(CloudConfiguration::AzureGovernment) => "https://login.microsoftonline.us".to_string(),
+        Some(CloudConfiguration::AzureChina) => "https://login.chinacloudapi.cn".to_string(),
+        Some(CloudConfiguration::AzurePublic) => AZURE_PUBLIC_CLOUD.to_string(),
+        // need this arm because CloudConfiguration is non-exhaustive
+        _ => {
+            return Err(Error::with_message(
+                ErrorKind::Other,
+                format!("unexpected cloud configuration: {:?}", cloud),
+            ))
+        }
+    };
 
-    Url::parse(&authority_host).map_err(Into::<Error>::into)
+    let url = Url::parse(&authority_host).map_err(Into::<Error>::into)?;
```

Comment:
Tip: this should also work:

```suggestion
    let url = Url::parse(&authority_host)?;
```